server/jobData/jobData.py

```diff
@@ -0,0 +1,64 @@
+import os
+import boto3
+import json
+
+def getReportJson(event, context):
+    """
+    Input: POST request where the body is a stringified json of the form:
+    "{"publicKey":"1", "pwd":"password"}"
+    Where the values for publicKey and pwd are the ID and PASSWORD from link to display batch job report
+    Eg. https://afternoon-woodland-24079.herokuapp.com/batchreportpage/8944s61gdgds300187?pwd=blhTelphY
+        where ID = 8944s61gdgds300187, PASSWORD = blhTelphY
+
+    Output: presigned url to report.json for job in s3 bucket
+    """
+
+    # Load id/password from html request
+    json_body = json.loads(event["body"])
+    linkId = json_body["publicKey"]
+    linkPassword = json_body["pwd"]
+
+    # Constants
+    TABLE_NAME = os.environ["JOB_TABLE"]
+    BUCKET_NAME = os.environ["SRC_BUCKET"]
+    LOOKUP_BATCH_ID = linkId # may lookup batch job id in another table using linkId for extra security
+    PRESIGNED_LINK_DURATION = 60*5 # 5 minutes in seconds
+
+    # fetch from database
+    table = boto3.resource("dynamodb").Table(TABLE_NAME)
+    data = table.get_item(Key={"id": LOOKUP_BATCH_ID})
+
+    # check if job id exists
+    try:
+        data["Item"]
+    except KeyError:
+        return {
+            "statusCode": 404,  # job id not found
+            "body": "{}",
```

Comment:
might be helpful to add the error message to the returned body

Reply:
I also think this should be changed to return an informative error message.